chapters/en/slides/module6/module6_03.Rmd

```diff
@@ -0,0 +1,179 @@
+```
+
+Notes:
+On this slide we are recreating the choropleth and pointplot map
+from the end of the last slide deck.
+We are saving these as variables so that we can use them to practice
+laying out plots on the following slides.
+
+TODO fix so that this code shows up correctly on the slide
+
+---
+
+## Placing plots next to each other allows us to make more direct comparisons
+
+```{python}
+choropleth | points_on_map
+```
+
+Notes:
+To place plot side by side,
+vi can use the pipe (`|`) operator.
+You have already seen us using this occasionally in previous modules.
+
+We could also have used the `alt.hconcat(choropleth, points_on_map)`,
+but it takes less typing to use the operator.
+
+We can see that the legends are stacked to the right of both plots.
+
+---
+
+## Plots can also be concatenated vertically
+
+```{python}
+choropleth & points_on_map
+```
+
+Notes:
+To place plots next to each other vertically,
+we can use the ampersand (`&`) operator.
+
+We could also have used the `alt.vconcat(choropleth, points_on_map)`,
+but it takes less typing to use the operator.
+
+---
+
+## Vertical and horizontal concatenation can be combined
+
+```{python}
+bars = alt.Chart(state_pop).mark_bar().encode(
+    y='population',
+    x=alt.Y('state', sort='y', title=None),
+    color='population').properties(height=100, width=625)
+
+points_on_map | choropleth & bars
+```
+
+Notes:
+Often we want to create more complex figure layouts
+where plots are concatenated both horizontally and vertically.
+This allow to present relevant information next to each other
+and can be useful if there is a key comparison we want to make
+between multiple slots on a single slide
+instead of flipping between slides
+which can be harder to follow.
+
+In Altair,
+we can create the figure layout
+by combining the vertical and the horizontal concatenation operators.
+
+In this slide,
+we create a barplot of the population in each state
+that we want to place under the two maps.
+But as we can see in this slide,
+this places the barplot under only the rightmost map.
+Why is that?
+
+---
+
+## Parentheses can be used to indicate groupings in complex layouts
+
+```{python}
+bars = alt.Chart(state_pop).mark_bar().encode(
+    y='population',
+    x=alt.Y('state', sort='y', title=None),
+    color='population').properties(height=100, width=625)
+
+(points_on_map | choropleth) & bars
+```
+
+Notes:
+The reason the plot in the previous slide did not looks as we desired
+is that the `&` operator has higher priority than `|`,
+so the choropleth and barplot are first vertically concatenated
+before being horizontally concatenated to the point map.
+
+We want the horizontal concatenation between the maps 
+to occur first
+and just like with mathematical expression,
+we can use parenthesis to indicate 
+which operations should have priority.
+
+You can see an example of that in this slide,
+where we get the barplot to show up below both the maps
+as desired.
```

Comment:
Should we somewhere discuss why this version of the panel figure is more effective? We seem to assume students will just know that, but I don't think we have taught them about the effectiveness of panel figures.

```suggestion
The reason that the plot in the previous slide did not look like we wanted it to, 
was that the `&` operator has higher priority than `|` operator.

The resulted in the choropleth and barplot being vertically concatenated
before the point map was horizontally concatenated.

What we really want is the horizontal concatenation between the maps 
to occur first, followed by the vertical concatenation of the barplot.

Just as we use parentheses in mathematics to change the default order 
of mathematical operations, we can use parentheses to change the default order 
of concatenation operations.

You can see an example of that in this slide,
where we place the parentheses around the `points_on_map | choropleth`
to force the horizontal concatenation between the maps 
to occur first, and that leads us to get the barplot to show up below both the maps
as desired.
```